google/cloud/aiplatform/training_utils/cloud_profiler/plugins/tensorflow/tf_profiler.py

```diff
@@ -17,6 +17,15 @@
 
 """A plugin to handle remote tensoflow profiler sessions for Vertex AI."""
 
+try:
+    from tensorboard_plugin_profile.profile_plugin import ProfilePlugin
+    from werkzeug import Response
```

Comment:
`initializer.py` imports `werkzeug` before importing `tf_profiler`: https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/training_utils/cloud_profiler/initializer.py#L21 

Recommend catching on all optional imports or walking through all the imports to determine when they are first imported.

Reply:
Done, I went with catching the imports at where they were first imported. The tests should catch the verbiage as well.